backend/src/main/java/com/bakdata/conquery/resources/admin/rest/AdminProcessor.java

```diff
@@ -156,10 +156,10 @@ public synchronized Dataset addDataset(String name) throws JSONException {
 		dataset.setName(name);
 
 		// store dataset in own storage
-		NamespaceStorage datasetStorage = new NamespaceStorageImpl(
-			storage.getValidator(),
-			config.getStorage(),
-			new File(config.getStorage().getDirectory(), "dataset_" + name));
+		NamespaceStorage datasetStorage = config.getStorage().createNamespaceStorage(storage.getValidator(), List.of(storagePrefix, "dataset_" + name), true);
+		if (datasetStorage == null ) {
```

Comment:
Fehlermeldung ist falsch

Reply:
Wie meinst du das? Null bedeutet doch in dem Fall das der Storage schon existiert

Reply:
Echt? Ich hätte gedacht, dass das dann nicht existiert?

Reply:
Dafür habe ich der Methode hinten das Flag hinzugefügt. Wie würdest du es machen?

Reply:
Ich schmeiße jetzt einfach eine Exception, wenn der Ordner schon vorhanden ist der von dem Environment kreiert werden würde